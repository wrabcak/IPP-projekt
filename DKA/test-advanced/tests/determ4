# priklad http://www.cs.odu.edu/~toida/nerzic/390teched/regular/fa/nfa-2-dfa.html
( {s0, s1, S2, s3, s4}, 
  {'a', 'b'}, 
  {
    s0 'a' -> s1, 
    s0 'a' -> S2, 
    s0 'b' -> S2,
    s0 'a' -> s3,
    s0 'b' -> s3,
    s1 'a' -> s1,
    s1 'a' -> S2,
    s1 'b' -> S2,
    s1 'b' -> s3,
    S2 'b' -> S2,
    S2 'b' -> s4,
    S2 'b' -> s3,    
    s3 'b' -> S2,
    s3 'b' -> s3,
    s3 'a' -> s4,
    s3 'b' -> s4
  },
  s0, 
  {s1}
)
# Ocekavany vystup:
# s0 'a' -> s1_S2_s3
# s0 'b' -> S2_s3
# s1_S2 'a' -> s1_S2 
# s1_S2 'b' -> S2_s3_s4
# s1_S2_s3 'a' -> s1_S2_s4
# s1_S2_s3 'b' -> S2_s3_s4
# s1_S2_s4 'a' -> s1_S2
# s1_S2_s4 'b' -> S2_s3_s4
# S2_s3 'a' -> s4
# S2_s3 'b' -> S2_s3_s4
# S2_s3_s4 'a' -> s4
# S2_s3_s4 'b' -> S2_s3_s4
