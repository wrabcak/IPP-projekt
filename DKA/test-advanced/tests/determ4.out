(
{S2_s1, S2_s1_s3, S2_s1_s4, S2_s3, S2_s3_s4, s0, s4},
{'a', 'b'},
{
S2_s1 'a' -> S2_s1,
S2_s1 'b' -> S2_s3_s4,
S2_s1_s3 'a' -> S2_s1_s4,
S2_s1_s3 'b' -> S2_s3_s4,
S2_s1_s4 'a' -> S2_s1,
S2_s1_s4 'b' -> S2_s3_s4,
S2_s3 'a' -> s4,
S2_s3 'b' -> S2_s3_s4,
S2_s3_s4 'a' -> s4,
S2_s3_s4 'b' -> S2_s3_s4,
s0 'a' -> S2_s1_s3,
s0 'b' -> S2_s3
},
s0,
{S2_s1, S2_s1_s3, S2_s1_s4}
)